python/lsst/pipe/analysis/plotUtils.py

```diff
@@ -480,3 +522,82 @@ def getQuiver(x, y, e1, e2, ax, color=None, scale=3, width=0.005, label=''):
     q = ax.quiver(x, y, c1, c2, color=color, angles='uv', scale=scale, units='width', pivot='middle',
                   width=width, headwidth=0.0, headlength=0.0, headaxislength=0.0, label=label)
     return q
+
+
+def makeAlphaCmap(cmap=plt.cm.viridis, alpha=1.0):
+    """Given a matplotlib colormap, return it but with given alpha transparency
+
+    Parameters
+    ----------
+    cmap : `matplotlib.colors.ListedColormap`, optional
+       The matplotlib colormap to make transparent with level ``alpha``.
+       Default color map is `plt.cm.viridis`.
+    alpha : `float`, optional
+       The matplotlib blending value, between 0 (transparent) and 1 (opaque)
+       (1.0 by default).
+
+    Returns
+    -------
+    alphaCmap : `matplotlib.colors.ListedColormap`
+       The matplotlib colormap ``cmap`` but with transparency level ``alpha``.
+    """
+    alphaCmap = cmap(np.arange(cmap.N))
+    alphaCmap[:, -1] = alpha
+    alphaCmap = ListedColormap(alphaCmap)
+    return alphaCmap
+
+
+def buildTractImage(butler, dataId, tractInfo, patchList=None, coaddName="deep"):
+    """Build up an image of an entire tract or list of patches
+
+    Parameters
+    ----------
+    butler : `lsst.daf.persistence.Butler`
+    dataId : `lsst.daf.persistence.DataId`
+       An instance of `lsst.daf.persistence.DataId` from which to extract the
+       filter name.
+    tractInfo : `lsst.skymap.tractInfo.ExplicitTractInfo`
+       Tract information object.
+    patchList : `list` of `str`, optional
+       A list of the patches to include.  If `None`, the full list of patches
+       in ``tractInfo`` will be included.
+    coaddName : `str`, optional
+       The base name of the coadd (e.g. "deep" or "goodSeeing").
+       Default is "deep".
+
+    Raises
+    ------
+    `RuntimeError`
+       If ``nPatches`` is zero, i.e. no data was found.
+
+    Returns
+    -------
+    image : `lsst.afw.image.ImageF`
+       The full tract or patches in ``patchList`` image.
+    """
+    tractBbox = tractInfo.getBBox()
+    nPatches = 0
+    if not patchList:
+        patchList = []
+        nPatchX, nPatchY = tractInfo.getNumPatches()
+        for iPatchX in range(nPatchX):
+            for iPatchY in range(nPatchY):
+                patchList.append("%d,%d" % (iPatchX, iPatchY))
+    tractArray = np.empty((tractBbox.getMaxY() + 1, tractBbox.getMaxX() + 1), dtype="float32")
```

Comment:
This works too: `tractArray = np.full((tractBbox.getMaxY() + 1, tractBbox.getMaxX() + 1), np.nan, dtype="float32")`